File: elixir/line-up/lib/line_up.ex
defmodule LineUp do
  def format(name, number), do:
    "#{name}, you are the #{number}#{suffix(number)} customer we serve today. Thank you!"
  defp suffix(n) when rem(n, 10) == 1 and rem(n, 100) != 11, do: "st"
  defp suffix(n) when rem(n, 10) == 2 and rem(n, 100) != 12, do: "nd"
  defp suffix(n) when rem(n, 10) == 3 and rem(n, 100) != 13, do: "rd"
  defp suffix(_), do: "th"
end

File: csharp/line-up/LineUp.cs
public static class LineUp
{
    public static string Format(string name, int number) =>
        $"{name}, you are the {number}{Suffix(number)} customer we serve today. Thank you!";
    
    private static string Suffix(int x) => x switch
    {
        int n when n % 10 == 1 && n % 100 != 11 => "st",
        int n when n % 10 == 2 && n % 100 != 12 => "nd",
        int n when n % 10 == 3 && n % 100 != 13 => "rd",
        _ => "th"
    };
}
